src/DynamoPublish/Models/PublishModel.cs

```diff
@@ -0,0 +1,131 @@
+﻿using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.Nodes;
+using Dynamo.Wpf.Authentication;
+using Greg;
+using Greg.AuthProviders;
+using Reach;
+using Reach.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Dynamo.Publish.Models
+{
+    public class PublishModel
+    {
+        private readonly IAuthProvider authenticationProvider;
+        private readonly ICustomNodeManager customNodeManager;
+
+        private readonly string serverUrl;
+        private readonly string port;
+        private readonly string page;
+
+        private IWorkspaceStorageClient reachClient;
+
+        public bool IsLoggedIn
+        {
+            get
+            {
+                return authenticationProvider.LoginState == LoginState.LoggedIn;
+            }
+        }
+
+        public HomeWorkspaceModel HomeWorkspace
+        {
+            get;
+            private set;
+        }
+
+        public List<CustomNodeWorkspaceModel> CustomNodesWorkspaces
+        {
+            get;
+            private set;
+        }
+
+        #region Initialization
+
+        public PublishModel(IAuthProvider dynamoAuthenticationProvider, ICustomNodeManager dynamoCustomNodeManager)
+        {
+            // Open the configuration file using the dll location.
+            var config = ConfigurationManager.OpenExeConfiguration(this.GetType().Assembly.Location);
+            // Get the appSettings section.
+            var appSettings = (AppSettingsSection)config.GetSection("appSettings");
+
+            serverUrl = appSettings.Settings["ServerUrl"].Value;
+            if (String.IsNullOrWhiteSpace(serverUrl))
+                throw new ArgumentException();
+
+            port = appSettings.Settings["Port"].Value;
+            if (String.IsNullOrWhiteSpace(port))
+                throw new ArgumentException();
+
+            page = appSettings.Settings["Page"].Value;
+            if (String.IsNullOrWhiteSpace(page))
+                throw new ArgumentException();
+
```

Comment:
An `ArgumentException` is the wrong kind of exception to throw here. I recommend throwing a general exception with a good description.  For example, "The setting 'Port' was not found in the configuration file for DynamoPublish."